File: Tellma/ClientApp/src/app/data/dto/entities-response.ts
import { EntityWithKey } from '../entities/base/entity-with-key';
import { Entity } from '../entities/base/entity';

export interface EntitiesResponse<TEntity extends Entity = EntityWithKey> {
  Extras: {
    [key: string]: any;
  };
  Result: TEntity[];
  CollectionName: string;
  RelatedEntities: {
    [key: string]: EntityWithKey[];
  };
  IsPartial: boolean;
  ServerTime: string;
}

File: Tellma.Api.Web.Dto/EntitiesResponse.cs
using System;
using System.Collections.Generic;
using Tellma.Model.Common;

namespace Tellma.Api.Dto
{
    /// <summary>
    /// Represents a response of raw entities for save.
    /// </summary>
    public class EntitiesResponse<TEntity> where TEntity : Entity
    {
        public Extras Extras { get; set; }

        public IEnumerable<TEntity> Result { get; set; }

        public string CollectionName { get; set; }

        public Dictionary<string, IEnumerable<Entity>> RelatedEntities { get; set; }

        public DateTimeOffset ServerTime { get; set; }
    }
}
